parquet/src/main/java/org/apache/iceberg/parquet/TypeWithSchemaVisitor.java

```diff
@@ -76,16 +81,22 @@
               element = list.fields().get(0);
             }
 
-            visitor.fieldNames.push(repeatedElement.getName());
+            if (!isOldListElementType) {
+              visitor.fieldNames.push(repeatedElement.getName());
+            }
```

Comment:
This needs newlines between control flow blocks. It also warrants a comment about why this is skipping the name handling.